sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/NonAzureOpenAIKeyCredential.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai.models;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Objects;
+
+/**
+ * Token credential class for accepting non-Azure OpenAI API key credential string.
+ */
+public class NonAzureOpenAIKeyCredential {
+    private static final ClientLogger LOGGER = new ClientLogger(NonAzureOpenAIKeyCredential.class);
+
+    private String key;
+
+    /**
+     * Create a non-Azure OpenAI API key credential.
+     *
+     * @param key non-Azure OpenAI API key.
+     * @throws NullPointerException If key is null.
+     * @throws IllegalArgumentException If key is an empty string.
+     */
+    public NonAzureOpenAIKeyCredential(String key) {
+        Objects.requireNonNull(key, "'key' cannot be null.");
+        if (key.isEmpty()) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'key' cannot be empty."));
+        }
+        this.key = key;
+    }
+
+    /**
+     * Retrieves the key associated to this credential.
+     *
+     * @return The key being used to authorize requests.
+     */
+    public String getKey() {
+        return this.key;
```

Comment:
How is the prefix handled? We need to add `Bearer ` as the prefix for every key. @alzimmermsft has a [PR to support prefixes](https://github.com/Azure/azure-sdk-for-java/pull/35010) for `AzureKeyCredentialPolicy`. We should do something similar here.

Reply:
updated